pkg/cloud/services/ec2/instances_test.go

```diff
@@ -2141,7 +2141,7 @@ func TestCreateInstance(t *testing.T) {
 							AvailabilityZone:    aws.String("us-east-1b"),
 							MapPublicIpOnLaunch: aws.Bool(true),
 						}},
-					}, nil)
+					}, nil).MaxTimes(2)
```

Comment:
Why set this max? To fail earlier?

Reply:
I believe the mock was failing due to it being called more than once.  I believe the intent here was address that since it is supposed to be called twice.

Reply:
As Richard said, this test was failing due missing call to `DescribeSubnetsWithContext`, when I set `MaxTimes(2)` it was fixed. Now I am revising those tests and it seems the `DescribeSubnetsWithContext` is duplicated, and this change was recently added. I am not gomock expert, but removing the duplicated call ref resolved those issues (duplicated definition and MaxTimes). See the changes in file [pkg/cloud/services/ec2/instances_test.go](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4905/files#diff-3a3c262a621538f386d0529f1d3089210c0b205e0601318cca26e2804b427185)